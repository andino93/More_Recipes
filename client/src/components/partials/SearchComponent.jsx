import React, { Component } from 'react';
import PropTypes from 'react-proptypes';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { searchRecipe } from '../../actions/recipe/recipeActions';


/**
 * @export
 * 
 * @class SearchComponent
 * 
 * @extends {React.Component}
 */
export class SearchComponent extends Component {
  /**
   *
   * @param {props} props
   */
  constructor(props) {
    super(props);
    this.state = {
      searchParams: ''
    };
    this.handleSearch = this.handleSearch.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }
  /**
   * Handles Search
   *
   * @param {string} searchParams
   *
   * @method handleSearch
   *
   * @return {void}
   */
  handleSearch() {
    const { searchParams } = this.state;
    if (searchParams.length > 3) {
      this.props.searchRecipe(searchParams);
    }
  }
  /**
   * Handle change
   * 
   * @param {event} event
   *
   * @method handleChange
   *
   * @return {void}
   */
  handleChange(event) {
    this.setState({ searchParams: event.target.value });
  }
  /**
     * 
     * @memberof SearchComponent
     * 
     * @returns {XML} XML/JSX
     * 
     */
  render() {
    return (
      <div className="container-fluid banner pt-90">
        <div className="row">
          <div className="col-lg-12 text-white">
            <h2 className="text-center">More Recipe</h2>
            <p
              className="lead text-center"
              style={{
                opacity: 0.9
              }}>
              Amazing food recipes at your beck</p>

            <div id="custom-search-input center-content">
              <div className="input-group col-md-6 mx-auto">
                <input
                  type="text"
                  className="search-query form-control form-control-lg"
                  placeholder="Search recipe"
                  name="searchParams"
                  onChange={this.handleChange}
                />

                <span className="input-group-btn">
                  <button className="btn btn-outline-warning btn-lg"
                    type="button"
                    disabled={this.state.searchParams.length < 3}
                    onClick={this.handleSearch}
                  >
                    <span className="fa fa-search" />
                  </button>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

SearchComponent.propTypes = {
  searchRecipe: PropTypes.func,
  searchState: PropTypes.object
};

const mapStateToProps = state => ({
  searchState: state.recipeReducer
});

const mapDispatchToProps = dispatch =>
  bindActionCreators({ searchRecipe }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(SearchComponent);
