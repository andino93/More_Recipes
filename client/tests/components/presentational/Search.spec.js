import React from 'react';
import expect from 'expect';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { configure } from 'enzyme';
import mockStore from '../../__mocks__/mockStore';
import mountWrap from '../../__mocks__/utils';
import SearchComponent from '../../../src/components/partials/SearchComponent.jsx';

configure({ adapter: new Adapter() });

describe('<SearchComponent>', () => {
  const store = mockStore({});
  const connectedSearch = mountWrap(SearchComponent, store, {});
  it('should render the SearchComponent', (done) => {
    connectedSearch.find('input').at(0).simulate('change', {
      target: {
        value: 'Something'
      }
    });
    expect(connectedSearch.find('SearchComponent').instance().state.searchParams).toEqual('Something');
    expect(toJson(connectedSearch)).toMatchSnapshot();
    done();
  });
  it('should simulate handleSearch Method', (done) => {
    connectedSearch.find('button').at(0).simulate('click');
    done();
  });
});
