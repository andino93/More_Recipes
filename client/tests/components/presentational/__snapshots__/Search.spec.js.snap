// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchComponent> should render the SearchComponent 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "blank",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(SearchComponent)>
        <SearchComponent
          searchRecipe={[Function]}
        >
          <div
            className="container-fluid banner pt-90"
          >
            <div
              className="row"
            >
              <div
                className="col-lg-12 text-white"
              >
                <h2
                  className="text-center"
                >
                  More Recipe
                </h2>
                <p
                  className="lead text-center"
                  style={
                    Object {
                      "opacity": 0.9,
                    }
                  }
                >
                  Amazing food recipes at your beck
                </p>
                <div
                  id="custom-search-input center-content"
                >
                  <div
                    className="input-group col-md-6 mx-auto"
                  >
                    <input
                      className="search-query form-control form-control-lg"
                      name="searchParams"
                      onChange={[Function]}
                      placeholder="Search recipe"
                      type="text"
                    />
                    <span
                      className="input-group-btn"
                    >
                      <button
                        className="btn btn-outline-warning btn-lg"
                        disabled={false}
                        onClick={[Function]}
                        type="button"
                      >
                        <span
                          className="fa fa-search"
                        />
                      </button>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </SearchComponent>
      </Connect(SearchComponent)>
    </Router>
  </BrowserRouter>
</Provider>
`;
